futures-util/src/future/abortable.rs

```diff
@@ -155,14 +159,20 @@ impl<Fut> Future for Abortable<Fut> where Fut: Future {
         // registration.
         // Checking with `Relaxed` is sufficient because `register` introduces an
         // `AcqRel` barrier.
-        if self.inner.cancel.load(Ordering::Relaxed) {
+        if self.is_aborted() {
             return Poll::Ready(Err(Aborted))
         }
 
         Poll::Pending
     }
 }
 
+impl<Fut> FusedFuture for Abortable<Fut> where Fut: FusedFuture {
+    fn is_terminated(&self) -> bool {
+        self.is_aborted() || self.future.is_terminated()
```

Comment:
I'm not sure if this is what we want-- the future may be aborted but not yet have yielded a value indicating that it has been aborted.